### ISSUE TITLE
TypeError: Cannot read property 'id' of undefined

### ISSUE THREAD

**Issue by Lachlann** [Joniff/Terratype#3]
Hello,
I have had this property working however recently the map fails to load on my pages. I get a console error of:
`TypeError: Cannot read property 'id' of undefined
    at Object.loadEditor (terratype.js?cache=1.0.10&cdv=795778288:830)
    at Object.initEditor (terratype.js?cache=1.0.10&cdv=795778288:932)
    at angular.min.js?cdv=795778288:74
    at a.$eval (angular.min.js?cdv=795778288:92)
    at pre (angular.min.js?cdv=795778288:158)
    at k (angular.min.js?cdv=795778288:44)
    at e (angular.min.js?cdv=795778288:40)
    at angular.min.js?cdv=795778288:39
    at archetype.js?cdv=795778288:1417
    at angular.min.js?cdv=795778288:100`

I have version 1.0.10 installed.

I am running Umbraco version 7.5.11 assembly: 1.0.6274.27352 and the map property is beign used within Archetype. 

I have a key entered which is valid and the preview map shown in the property settings is showing. However in the editor  I see: 'There was an error loading this map'
The json value for the example map i am trying to load is:

`{
  "zoom": 13,
  "position": {
    "id": "WGS84",
    "datum": "55.843314,-4.429961"
  },
  "lookup": "High St, Paisley PA1 2BE, UK"
}`

Any ideas?

**Comment by Joniff** [comment]
Hi Lachlann,

I'm guessing, looking at the line that caused your error, that you have no label enabled, and while there should be a label check before that code, there isn't. This is something that I will need to fix, so I'm very grateful for you raising this.

I suspect as a fix for you, your end, if you enable labels for your Terratype datatype, select 'Standard' as the label style. Save your datatype, then return to the Labels and disable them, and press save again. This will mean that even though your not using labels, the labels will have a style of 'Standard' in the config.

I know it looks like I'm asking you to set something, save it, then unset it and save it. But this hopefully will help, until I get a new version released.

![terratype standard label](https://cloud.githubusercontent.com/assets/12765209/24411148/05c12c98-13cd-11e7-858f-b779913db16c.png)

Cheers

**Comment by Lachlann** [comment]
Perfect, that has solved the problem thank you so much for the speedy reply.

**Comment by kgiszewski** [comment]
Same thing just happened to me except the fix was to pick a coordinate system. Maybe it can default to one?

**Comment by Joniff** [comment]
Agreed, I need to put some more validation when creating the data type. As, not having a coordinate system defined, is a bit fatal for a map. Will update roadmap to include this.
